src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -885,6 +886,7 @@ def call_from_extension() -> bool:
     """Return true if current request is from extension."""
     from promptflow._core.operation_context import OperationContext
 
+    update_user_agent_from_env_var()
```

Comment:
Upon reviewing the implementation of this function, I understand the rationale behind the logic for deleting the user agent. However, I recommend refining the approach in `update_user_agent_from_env_var()`. We should first check if the user agent from the environment is already present in `OperationContext`. If it is, we can skip the update; if not, proceed with appending it. This could streamline the logic and avoid unnecessary deletions.

Reply:
I will roll back the code first to catch this release.
New modifications will be made later.